src/qt/pivx/masternodeswidget.cpp

```diff
@@ -263,18 +262,17 @@ bool MasterNodesWidget::startMN(CMasternodeConfig::CMasternodeEntry mne, std::st
 
 void MasterNodesWidget::onStartAllClicked(int type)
 {
-    WalletModel::UnlockContext ctx(walletModel->requestUnlock());
-    if (!ctx.isValid()) {
-        // Unlock wallet was cancelled
-        inform(tr("Cannot perform Mastenodes start, wallet locked"));
+    std::unique_ptr<WalletModel::UnlockContext> pctx = MakeUnique<WalletModel::UnlockContext>(walletModel->requestUnlock());
```

Comment:
Would be good to move this inside the else block of the `isLoading`.  If the process in being performed, we would be unlocking the wallet under this context for nothing.

Reply:
Yes, good idea.